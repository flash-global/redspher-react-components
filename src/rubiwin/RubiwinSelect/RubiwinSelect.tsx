import Select from '@mui/material/Select';
import { SelectProps } from '@mui/material/Select/Select';
import RubiwinCaretBottomIcon from '../../icons/Rubiwin/CaretBottomIcon';
import themeConstants from '../theme/themeConstants';

interface IrubiwinSelect extends SelectProps {
  autoWidth?: boolean;
  children?: Node;
  className?: string;
  id?: string;
  inputProps?: object;
  multiple?: boolean;
  open?: boolean;
  value?: any;
}

/**
 * This is a Mui Select branded for Rubiwin
 *
 * Demos:
 * - [Selects](https://mui.com/components/selects/)
 *
 * API:
 * - [Select API](https://mui.com/api/select/)
 * - inherits [OutlinedInput API](https://mui.com/api/outlined-input/)
 */
const RubiwinSelect = ({
  children,
  className,
  autoWidth,
  id,
  inputProps,
  multiple,
  open,
  value,
  ...props
}: IrubiwinSelect) => (
  <Select
    autoWidth={autoWidth}
    id={id}
    inputProps={inputProps}
    multiple={multiple}
    open={open}
    value={value}
    className={className}
    IconComponent={(props) => (
      <RubiwinCaretBottomIcon
        primarycolor={
          /MuiSelect-iconOpen/.test(props?.className)
            ? themeConstants.primary.main
            : themeConstants.grey.main
        }
        {...props}
      />
    )}
    {...props}
  >
    {children}
  </Select>
);

export default RubiwinSelect;
