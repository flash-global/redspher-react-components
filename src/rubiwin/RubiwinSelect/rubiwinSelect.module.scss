@import '../rubiwin-constant';

.select {
  &:focus {
    background-color: rgba($primaryColor, 0.05) !important;
  }
}
