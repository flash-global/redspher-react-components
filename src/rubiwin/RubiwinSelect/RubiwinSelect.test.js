import renderer from 'react-test-renderer';
import RubiwinSelect from './RubiwinSelect';

const testProps = {
  className: 'test',
  autoWidth: false,
  id: 'testId',
  onChange: () => '',
  onClose: () => '',
  onOpen: () => '',
  open: true,
  renderValue: () => '',
  value: ''
};

it('renders correctly props object', () => {
  const tree = renderer.create(<RubiwinSelect {...testProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('is multiple', () => {
  const tree = renderer.create(<RubiwinSelect multiple />).toJSON();
  expect(tree).toMatchSnapshot();
});
