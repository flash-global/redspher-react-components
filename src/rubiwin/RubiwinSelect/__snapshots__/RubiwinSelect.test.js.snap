// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is multiple 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <div
    aria-haspopup="listbox"
    className="MuiSelect-root MuiSelect-select select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8203;",
        }
      }
    />
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;

exports[`renders correctly props object 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline test"
  onClick={[Function]}
>
  <div
    aria-haspopup="listbox"
    aria-labelledby="testId"
    className="MuiSelect-root MuiSelect-select select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    id="testId"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8203;",
        }
      }
    />
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value=""
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
